File: 프로그래머스/lv2/42626. 더 맵게/더 맵게.java
import java.util.PriorityQueue;

class Solution {
    public int solution(int[] scoville, int K) {
        int answer = 0;
        PriorityQueue<Integer> pq=new PriorityQueue();
        for(int s:scoville){
            pq.offer(s);
        }

        while(pq.peek()<=K){
            if(pq.size()==1){
                return -1;
            }
            
            int a=pq.poll();
            int b=pq.poll();
            
            int tmp=a+(b*2);
            
            pq.offer(tmp);
            answer++;
        }
        
        return answer;

    }
}

File: 프로그래머스/2/42626. 더 맵게/더 맵게.py
import heapq

def solution(scoville, K):
    answer = 0
    heap = []

    for s in scoville:
        heapq.heappush(heap, s)
    while heap[0] < K:
        if len(heap) == 1:
            return -1

        n = heapq.heappop(heap)
        t = heapq.heappop(heap)
        heapq.heappush(heap, n + (t * 2))
        answer += 1

    return answer
